pkg/webservice/webservice.go

```diff
@@ -41,26 +42,29 @@ type WebService struct {
 	httpServer *http.Server
 }
 
-func newRouter() *mux.Router {
-	router := mux.NewRouter().StrictSlash(true)
+func newRouter() *httprouter.Router {
+	router := httprouter.New()
 	for _, webRoute := range webRoutes {
 		handler := loggingHandler(webRoute.HandlerFunc, webRoute.Name)
-		router.
-			Methods(webRoute.Method).
-			Path(webRoute.Pattern).
-			Name(webRoute.Name).
-			Handler(handler)
+		switch webRoute.Method {
+		case "GET":
+			router.HandlerFunc(http.MethodGet, webRoute.Pattern, handler)
+		case "PUT":
+			router.HandlerFunc(http.MethodPut, webRoute.Pattern, handler)
+		case "POST":
+			router.HandlerFunc(http.MethodPost, webRoute.Pattern, handler)
+		}
 	}
 	return router
 }
 
-func loggingHandler(inner http.Handler, name string) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+func loggingHandler(inner http.Handler, name string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
 		start := time.Now()
 		inner.ServeHTTP(w, r)
 		log.Logger().Debug(fmt.Sprintf("%s\t%s\t%s\t%s",
 			r.Method, r.RequestURI, name, time.Since(start)))
```

Comment:
Remove the Sprintf. Building the string causes overhead we do not want/need. Move to standard logging call using zap.String(name, value) for each, ordering: name, method, URI and time.Since(start)

Reply:
Good suggestion!